OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -644,6 +644,71 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 						node.Key = "ReloadDelay";
 				}
 
+				// Migrated several properties to use int percentages instead of float
+				if (engineVersion < 20160323)
+				{
+					if (node.Key == "SelfHealing")
+					{
+						var healIfBelowNode = node.Value.Nodes.FirstOrDefault(x => x.Key == "HealIfBelow");
+						if (healIfBelowNode != null)
+						{
+							// The HealIfBelow value is now an int percentage, so multiply the float with 100.
+							var oldValue = FieldLoader.GetValue<float>("HealIfBelow", healIfBelowNode.Value.Value);
+							healIfBelowNode.Value.Value = (oldValue * 100).ToString();
```

Comment:
You should convert this to an int to truncate any fractional part before turning it to a string.